packages/react/package.json

```diff
@@ -5,6 +5,22 @@
   "license": "Apache-2.0",
   "main": "lib/index.js",
   "module": "es/index.js",
+  "exports": {
+    ".": {
+      "node": "./lib/index.js",
```

Comment:
Hey Joe! 👋 

This is super random so feel free to disregard 😂 Was just working on this with something else and wanted to share.

You might be able to get away with only `import` and `require`. From what I understand, `node` is the most specific and will match in the Node.js environment even if calling from an ESM context (`"type": "module"`). Switching over to `import` and `require` allow you to retain the import semantics (ESM or CommonJS) regardless of environment (Node.js, bundler)

Hope I'm understanding that right 🤔 Would love to hear what you think!

Reply:
Hey @joshblack!!

Alright, so you got me digging deeper into this lol, but I'll also give you a little backstory as well.

My read of the docs was definitely the same as yours... `import` and `require` *should be sufficient*, and you can include `default` as a best practice for ultra portability. (`default` is also the one that the css-loader seems to pick up in webpack. It's possible it'd also use one of the higher-priority keys, but it seems to make more sense to keep the scss import in the default block instead of one that's really meant for ES/Node.)

But in practice, this was not behaving as expected 🤔. I get errors like this:

```
SyntaxError: The requested module '@carbon/react' does not provide an export named 'Accordion'
```

I can verify that `ts-node` grabs the ES import that it finds from the `exports` key, but because the es file's extension inside of `@carbon/react` isn't `mjs` and because the package isn't marked as `"type": "module"`, ts-node appears to treat this file as a commonjs module instead of an ES module. It's basically using Node to load the module (accepting ES6-style import syntax along the way), but then resolving its imports as commonjs-style imports when the module is used in a different file 🤷‍♂️.

Things move further if I set `"type": "module"` in the `carbon/react` pacakge.json or I start changing file extensions to `.mjs` (and I exclude the `"node":...` from the exports).

I'm actually not sure whose problem this is to solve 😂. I found this in [the proposal](https://github.com/jkrems/proposal-pkg-exports/) for the exports functionality:

> Resolution should not depend on file extensions, allowing ESM syntax in .js files.

But does that mean that those files are *treated* as ESM even if the package isn't marked as `"type": "module"`? The answer is actually "no" lol. The ES-style syntax is *allowed* in those files, but the module itself is still considered as a cjs module. Here's an example from a minimum repro:

<img width="1829" alt="image" src="https://user-images.githubusercontent.com/51208233/197316001-25cac047-67d0-42c0-a564-d5275e888dc9.png">

All of this is to say that because the chain of ES module files within `carbon/react` aren't **Node-compatible** ES modules (i.e. .mjs file extensions or package.json files with type:node in them), they won't work as-is when imported into a Nodejs environment, but they **will** still work in a browser environment, and there's probably some bundlers/runtimes that'll tolerate the discrepancy and figure out what to do. node/ts-node just isn't one of those.

Since I have a workaround, I think I'll close this PR for now and investigate my options for avoiding ts-node